### DISCUSSION TITLE
Incorrect redirects behind TLS termination proxy

### DISCUSSION THREAD

**Discussion by wolfdn** [apache/airflow#20666] [Q&A] [answered]
Hello! We have a problem with our Airflow instance running behind a TLS-terminating Azure Application Gateway. This is an overview of our setup:

![grafik](https://user-images.githubusercontent.com/95075445/148176398-053e5617-5fce-4ec6-831e-24be47fe1e7b.png)

To deploy Airflow on our AKS cluster, we use the official Airflow Helm chart.
We followed the instructions on the Airflow website for running Airflow behind a reverse proxy: https://airflow.apache.org/docs/apache-airflow/stable/howto/run-behind-proxy.html
We applied the mentioned settings to the ingress-nginx ingress controller (had to translate them to ingress-nginx annotations):
```
ingress:
  web:
    annotations: {
      nginx.ingress.kubernetes.io/ssl-redirect: "false",
      nginx.ingress.kubernetes.io/server-snippet: "proxy_set_header Host $http_host;\nproxy_redirect off;\nproxy_http_version 1.1;\nproxy_set_header Upgrade $http_upgrade;\nproxy_set_header Connection \"upgrade\";"
    }
```
And we applied the mentioned configuration to the Airflow webserver:
```
config:
  webserver:
    enable_proxy_fix: "True"
    base_url: https://airflow.mycompany.com
```
We now face the problem that Airflow doesn’t seem to be aware that it should redirect to https URLs. We observe the following behavior:
* When trying to access Airflow via the URL https://airflow.mycompany.com, we are redirected to **http**://airflow.mycompany.com/home and finally receive a timeout error
* When accessing Airflow via this URL: https://airflow.mycompany.com/home we arrive at the Airflow Sign In page, but it tries to make redirects to non-https URLs. This is visible for example in the URL of the Sign In page already: https://airflow.mycompany.com/login/?next=http%3A%2F%2Fairflow.mycompany.com%2Fhome (note the **http** in the next url) – trying to sign in results in a timeout error again

The documentation mentions that we need to make sure that the proxy sets the “X-Forwarded-Proto” header. We tried already telling ingress-nginx explicitly to do this using this server-snippet:
```
proxy_set_header X-Forwarded-Proto https;
```
But Airflow still tries to redirect to non-https URLs.

How can we tell Airflow that it should redirect to https URLs?

**Comment by wolfdn** [comment, marked as answer]
Okay so after struggling with this for quite a while, we finally found the solution:

We just had to enable the "use-forwarded-headers" option in the ingress-nginx configmap:
https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#use-forwarded-headers

So the problem was probably that ingress-nginx did not pass the X-Forwarded-Proto headers to Airflow.

**Reply by potiuk** [reply]
Good one - great to have it here as guide to others struggling with it!
